lib/internal/fs/promises.js

```diff
@@ -273,7 +276,21 @@ async function fsCall(fn, handle, ...args) {
 }
 
 async function writeFileHandle(filehandle, data, signal) {
-  // `data` could be any kind of typed array.
+  if (isIterable(data)) {
+    if (signal?.aborted) {
+      throw lazyDOMException('The operation was aborted', 'AbortError');
+    }
```

Comment:
To avoid creating the `UInt8Array`:
```suggestion
  if (signal?.aborted) {
    throw lazyDOMException('The operation was aborted', 'AbortError');
  }
  if (isIterable(data)) {
```